go/saltpackKeyHelpers/saltpack_pseudonym_resolvers.go

```diff
@@ -0,0 +1,176 @@
+package saltpackKeyHelpers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/teams"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"github.com/keybase/saltpack"
+)
+
+// This file contains two implementations of saltpack.SymmetricKeyResolver, which can be used to resolve
+// respectively the old Kbfs Pseudonyms and the newer Key Pseudonyms by querying the server. A mock implementation
+// (which does not communicate with the sever and avoids circular dependencies) is available in the saltpackHelperMocks package.
+
+// Resolves old kbfs pseudonyms.
+type TlfKeyResolver struct {
+	libkb.Contextified
+	ctx context.Context
+}
+
+var _ saltpack.SymmetricKeyResolver = (*TlfKeyResolver)(nil)
+
+func NewLegacyKBFSResolver(ctx context.Context, g *libkb.GlobalContext) saltpack.SymmetricKeyResolver {
```

Comment:
usually we don't store `context.Context` into objects, but instaed just pass it through.  Can we do that here? In other words, no arguments for this function, and just pass a `MetaContext` through.

Reply:
Here I switched from storing G and ctx to a MetaContext. 
It seems to me that in this case we need to store this MetaContext inside the object as the method ResolveKeys (which implements an interface defined in the saltpack repo and therefore unaware of our MetaContext) does not take such m (or any other type of Context) as input.